packages/vue-components/src/panels/PanelBase.js

```diff
@@ -99,76 +99,85 @@ export default {
-        as we need to know the scrollHeight of the content for the animation to work.
-        */
-        this.wasRetrieverLoaded = true;
-        this.$nextTick(() => {
-          this.localExpanded = true;
-        });
+      if (this.localExpanded) {
+        // COLLAPSE
+        // panel maxHeight is 'none' at this moment (as event listener set it to 'none' after expansion)
+        // thus, we need to reset the maxHeight to its current height (for collapse transition to work)
+        this.$refs.panel.style.maxHeight = `${this.$refs.panel.scrollHeight}px`;
+        // we need to reset / 'memorize' the expandedPanelHeight everytime we collapse
+        // this is because we may have expanded nested panels and the parent panel height is now different
+        this.expandedPanelHeight = this.$refs.panel.scrollHeight;
+        // need to wait for DOM update, after resetting panel maxHeight
+        setTimeout(() => {
+          // transition to collapsed panel
+          this.$refs.panel.style.maxHeight = `${this.collapsedPanelHeight}px`;
+        }, 0); // nextTick doesn't work, not sure why :O
       } else {
-        this.localExpanded = !this.localExpanded;
+        // EXPAND
+        this.$refs.panel.style.maxHeight = `${this.expandedPanelHeight}px`;
+        // we will always set maxHeight to 'none' after expansion
+        // this is to accomodate the expansion of nested panels (parent panel needs to grow)
+        // event listener will set maxHeight to none after transition ends for expansion
       }
+      this.localExpanded = !this.localExpanded;
     },
     close() {
       this.localExpanded = false;
       this.localMinimized = true;
+      // we need to reset / 'memorize' the expandedPanelHeight everytime we collapse
+      // this is because we may have expanded nested panels and the parent panel height is now different
+      this.expandedPanelHeight = this.$refs.panel.scrollHeight;
+      this.$refs.panel.style.maxHeight = `${this.collapsedPanelHeight}px`;
+      // we don't need transition for closing (change to minimized panel)
     },
     open() {
-      this.wasRetrieverLoaded = true;
-      this.$nextTick(() => {
-        this.localExpanded = true;
-      });
       this.localMinimized = false;
+      // need to wait for DOM update, after setting minimized to false
+      setTimeout(() => {
+        // transition to expanded
+        this.localExpanded = true;
+        this.$refs.panel.style.maxHeight = `${this.expandedPanelHeight}px`;
+      }, 0); // nextTick doesn't work, not sure why :O
     },
     openPopup() {
       window.open(this.popupUrl);
     },
-    setMaxHeight() {
-      // Don't play the transition for this case as the loading should feel 'instant'
-      if (this.expandedBool) {
-        this.$refs.panel.style.maxHeight = 'none';
+    setExpandedPanelHeight() {
+      this.expandedPanelHeight = this.$refs.panel.scrollHeight;
```

Comment:
hmm would this ever not be `this.$refs.panel.scrollHeight`? (also taking into account the panel preview pr)

Reply:
Thanks for bringing this up! I have removed `expandedPanelHeight` from my commit, since it is redundant, as you have pointed out.